tests/unit/backend/layers/business/test_business.py

```diff
@@ -0,0 +1,778 @@
+            Link("test doi", "doi", "http://bad.doi")
+        ]
+
+        self.sample_collection_metadata.links = links_with_doi
+
+        # TODO: make sure that we don't need different actions depending on which exception
+        self.crossref_provider.fetch_metadata = Mock(side_effect=CrossrefException("Error!"))
+
+        self.business_logic.create_collection(self.sample_collection_metadata, self.user_info)
+        # TODO: create CollectionCreationException
+        with self.assertRaises(CollectionCreationException):
+            self.business_logic.create_collection(self.sample_collection_metadata, self.user_info)
+
+    def test_create_collection_unauthorized_fail(self):
+        # TODO: AUTHORIZATION
+        return NotImplemented
+
+class TestDeleteCollection(BaseBusinessLogicTestCase):
+
+    def test_delete_collection_ok(self):
+        # TODO: verify how the current support for collection deletion is
+        return NotImplemented
+
+    def test_delete_collection_unauthorized_fail(self):
+        # TODO: AUTHORIZATION
+        return NotImplemented
+
+class TestGetCollection(BaseBusinessLogicTestCase):
+
+    def test_get_collection_published_ok(self):
+        """
+        A published collection can be obtained using `get_collection`
+        """
+        version = self.initialize_public_collection()
+        fetched_version = self.business_logic.get_collection(version.collection_id, self.user_info)
+        self.assertIsNotNone(fetched_version.published_at)
+        self.assertEqual(fetched_version.metadata, version.metadata)
+
+    def test_get_collection_unpublished_fail(self):
+        """
+        An unpublished collection should not be obtained using get_collection
+        Instead, `get_collection_version` should be used
+        """
+        version = self.initialize_private_collection()
+        fetched_version = self.business_logic.get_collection(version.collection_id, self.user_info)
+        self.assertIsNone(fetched_version)
+
+    def test_get_collection_version_ok(self):
+        """
+        A collection version can be obtained using `get_collection_version`
+        """
+        version = self.initialize_private_collection()
+        fetched_version = self.business_logic.get_collection_version(version.version_id, self.user_info)
+        self.assertIsNotNone(fetched_version.published_at)
+        self.assertEqual(fetched_version.metadata, version.metadata)
+
+class TestGetAllCollections(BaseBusinessLogicTestCase):
+
+    def test_setUp(self) -> None:
+        self.collection1 = self.initialize_private_collection(owner="test_user_1")
+        self.collection2 = self.initialize_private_collection(owner="test_user_2")
+        self.collection3 = self.initialize_public_collection(published_at=datetime(2022, 1, 1))
+        self.collection4 = self.initialize_public_collection(published_at=datetime(2021, 1, 1))
+        self.collection5 = self.initialize_public_collection(published_at=datetime(2020, 1, 1))
```

Comment:
Nit: I _much_ prefer inlining all test state setup code in the test itself and avoid having a multi-purpose test state.  This state is serving multiple purposes (owner filtering, published filtering, etc), and it will become harder to maintain a correct, shared test state as the tests grow.

Reply:
I'm not sure I'm following. This state is only pertinent to `TestGetAllCollections`, and it's immutable, so I don't see what can cause issues. Also, inlining the state in each test (rather than the unit) won't work because it will keep adding collections to the database. I apologize if I didn't get your concern.

Reply:
Suggested code changes on https://github.com/chanzuckerberg/single-cell-data-portal/pull/3505

Reply:
👍